docs/operator.md

```diff
@@ -0,0 +1,117 @@
+# Operator build and deployment
+
+## Prerequisites
+
+1. operator-sdk <https://sdk.operatorframework.io/docs/installation/>
+2. kustomize <https://github.com/kubernetes-sigs/kustomize/releases>
+3. opm <https://github.com/operator-framework/operator-registry/releases>
+
+## Building the operator
+
+### Create the bundle manifests
+
+```bash
+make operator-bundle
+```
+
+To generate the manifests and CSV in ./bundle, "make operator-bundle" first calls the "ocp-create-manifests" target. This target in turn calls "deploy-service-on-ocp-cluster and deploy-ui-on-ocp-cluster" while setting APPLY_MANIFESTS=False and APPLY_NAMESPACE=False. This causes the resource yamls to be created in ./build/assisted-installer/ but does not apply them.
+
+The relevant resource yamls are then copied to ./config/assisted-service where additional customizations are applied using Kustomize. The resulting yaml is then piped to operator-sdk
+creating the manifests and CSVs in ./bundle/manifests. 
+
+More information about bundles: <https://sdk.operatorframework.io/docs/olm-integration/generation/>
+
+### Create the bundle and index images
+
+```bash
```

Comment:
We usually use skipper, is there a target that can be called from the makefile? 
I don't want to mix between podman and the docker that we use in the makefile. it will be confusing.
It's not a blocker for this PR just something to think about.